File: ZfCustomView/app/src/main/java/zhanf/com/zfcustomview/activity/MainActivity.java
package zhanf.com.zfcustomview.activity;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import zhanf.com.zfcustomview.R;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.stv).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this, "onClick", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: ZfCustomView/app/src/main/java/zhanf/com/zfcustomview/main/activity/MainActivity.kt
package zhanf.com.zfcustomview.main.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_main.*
import zhanf.com.zfcustomview.R
import zhanf.com.zfcustomview.main.adapter.MainAdapter
import java.util.*

class MainActivity : AbstractActivity() {

    private val adapter by lazy { MainAdapter(this) }
    private val activityList = ArrayList<Class<out AppCompatActivity>>()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        activityList.add(SelectorTvActivity::class.java)
        activityList.add(MediaPlayerActivity::class.java)
        activityList.add(TransparentActivity::class.java)
        activityList.add(HistogramActivity::class.java)

        adapter.setActivityList(activityList)
        rvMain.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        rvMain.setHasFixedSize(true)
//        adapter.setHasStableIds(true)
        rvMain.adapter = adapter

    }
}
